### PR TITLE
feat: add json validation during file upload read buffer

### PR DESCRIPTION
**Pull request by mistahj67**
## Description

Adds json validation during the reading / writing of the stream buffer from file uploads and returns if an error is hit. 
This also removes the size limitation on the UI side.

## Motivation and Context

Currently there is little validation done until after the file is loaded and the UI does most of the file validation. We want to move the validation to server side.

## How Has This Been Tested?

An integration test using an image file was added

## Types of changes

-   [ ] Chore (a change that does not modify the application functionality)
-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

-   [ ] Documentation updates are needed, and have been made accordingly.
-   [x] I have added and/or updated tests to cover my changes.
-   [x] All new and existing tests passed.
-   [ ] My changes include a database migration.

### CODE REVIEW

File: cmd/api/src/api/v2/file_uploads_integration_test.go [hunk 1]
```diff
@@ -130,6 +130,16 @@ func Test_FileUpload(t *testing.T) {
 		assert.Nil(tx, err)
 		assert.Equal(tx, http.StatusUnsupportedMediaType, resp.StatusCode)
 	})
+
+	t.Run("not valid json", func(tx *testing.T) {
+		jsonInput := loader.GetReader("v6/ingest/SO_Logo.png")
```
Comment (mistahj67) [inline cmd/api/src/api/v2/file_uploads_integration_test.go:135]: I opted to toss an image in here to validate. Wasn't sure what image to use so went with logo, suggestions are appreciated!

File: cmd/api/src/api/v2/file_uploads.go [hunk 2]
```diff
@@ -128,12 +128,12 @@ func (s Resources) ProcessFileUpload(response http.ResponseWriter, request *http
 	} else if fileUploadJob, err := fileupload.GetFileUploadJobByID(s.DB, int64(fileUploadJobID)); err != nil {
 		api.HandleDatabaseError(request, response, err)
 	} else if fileName, err := fileupload.SaveIngestFile(s.Config.TempDirectory(), request.Body); err != nil {
-		if errors.Is(err, fileupload.ErrInvalidIngestFileType) {
-			api.WriteErrorResponse(request.Context(), api.BuildErrorResponse(http.StatusBadRequest, fmt.Sprintf("Error saving ingest file: %v", err), request), response)
+		if errors.Is(err, fileupload.ErrInvalidJSON) {
+			api.WriteErrorResponse(request.Context(), api.BuildErrorResponse(http.StatusHTTPVersionNotSupported, fmt.Sprintf("Error saving ingest file: %v", err), request), response)
```
Comment (superlinkx) [inline cmd/api/src/api/v2/file_uploads.go:132]: I don't think this was the intended error code

File: cmd/api/src/api/v2/file_uploads.go [hunk 3]
```diff
@@ -128,12 +128,12 @@ func (s Resources) ProcessFileUpload(response http.ResponseWriter, request *http
...
 		} else {
 			api.WriteErrorResponse(request.Context(), api.BuildErrorResponse(http.StatusInternalServerError, fmt.Sprintf("Error saving ingest file: %v", err), request), response)
 		}
-	} else if _, err = ingest.CreateIngestTask(s.DB, fileName, requestContext.RequestID, int64(fileUploadJobID)); err != nil {
+	} else if _, err = ingest.CreateIngestTask(s.DB, fileName, requestId, int64(fileUploadJobID)); err != nil {
```
Comment (superlinkx) [inline cmd/api/src/api/v2/file_uploads.go:136]: I know this is from before your changes, but we could make this flow a little nicer by doing:

```suggestion:
} else if fileName, err := fileupload.SaveIngestFile(s.Config.TempDirectory(), request.Body); if errors.Is(err, fileupload.ErrInvalidJSON) {
		api.WriteErrorResponse(request.Context(), api.BuildErrorResponse(http.StatusHTTPVersionNotSupported, fmt.Sprintf("Error saving ingest file: %v", err), request), response)
	} else if err != nil{
		api.WriteErrorResponse(request.Context(), api.BuildErrorResponse(http.StatusInternalServerError, fmt.Sprintf("Error saving ingest file: %v", err), request), response)
	} else if _, err = ingest.CreateIngestTask(s.DB, fileName, requestId, int64(fileUploadJobID)); err != nil {
```

File: cmd/api/src/api/v2/file_uploads_integration_test.go [hunk 4]
```diff
@@ -117,7 +117,7 @@ func Test_FileUpload(t *testing.T) {
 		assert.Nil(tx, err)
 		resp, err := apiClient.Raw(req)
 		assert.Nil(tx, err)
-		assert.Equal(tx, http.StatusBadRequest, resp.StatusCode)
+		assert.Equal(tx, http.StatusHTTPVersionNotSupported, resp.StatusCode)
```
Comment (superlinkx) [inline cmd/api/src/api/v2/file_uploads_integration_test.go:120]: Incorrect error code

File: cmd/api/src/api/v2/file_uploads_integration_test.go [hunk 5]
```diff
@@ -130,6 +130,16 @@ func Test_FileUpload(t *testing.T) {
...
 	})
+
+	t.Run("not valid json", func(tx *testing.T) {
+		jsonInput := loader.GetReader("v6/ingest/SO_Logo.png")
+		defer jsonInput.Close()
+		req, err := apiClient.NewRequest(http.MethodPost, jobEndpoint, nil, jsonInput)
+		assert.Nil(tx, err)
+		resp, err := apiClient.Raw(req)
+		assert.Nil(tx, err)
+		assert.Equal(tx, http.StatusHTTPVersionNotSupported, resp.StatusCode)
```
Comment (superlinkx) [inline cmd/api/src/api/v2/file_uploads_integration_test.go:141]: Incorrect error code

**Review by superlinkx — CHANGES_REQUESTED** [review]

File: cmd/api/src/api/v2/file_uploads_integration_test.go (hunk 1 above)
Comment (superlinkx) [reply cmd/api/src/api/v2/file_uploads_integration_test.go:135]: Seems fine

**Review by superlinkx — APPROVED** [review]

**Pull request merged**